src/time/Storage.cpp

```diff
@@ -163,10 +163,18 @@ Eigen::VectorXd Storage::sample(double time) const
 
   PRECICE_ASSERT(usedDegree >= 1);
 
-  auto data = getTimesAndValues();
+  auto [times, values] = getTimesAndValues();
+
+  //Return the sample corresponding to time if it exists
+  for (int i = 0; i < times.size(); i++) {
+    if (math::equals(times[i], time)) {
+      return values.col(i);
+    }
+  }
```

Comment:
This could be implemented more efficiently: `times` are ascending. Therefore, I would suggest something in this direction:
```
int i = 0;
while(math::smallerEquals(times[i], time) {
    if(math::equals(times[i], time)) {
        return values.col(i);
    }
    i++;
}
```
Maybe even put this into a private function of `Storage`:
```
int findTimeId(time) const {
    int i = 0;
    while(math::smallerEquals(_stampleStorage[i].timestamp, time) {
        if(math::equals(_stampleStorage[i].timestamp, time)) {
        return return i;
        }
    i++;
    }
    return -1;  // time not found in times
}
```

Reply:
If we start with efficiency here: You can move `getTimesAndValues()` inside the `if(!_bspline.has_value())` branch, because you don't need it outside. This results in

```
Eigen::VectorXd Storage::sample(double time) const {
    ...
    const int i = findTimeId(time);
    if(i > -1) {  // _stampleStorage contains sample at given time
        return _stampleStorage[i].sample;  // don't use getTimesAndValues, because this would iterate over the complete _stampleStorage.
    }

    if(!_bspline.has_value()) {
        auto [times, values] = getTimesAndValues();
        _bspline.emplace(times, values, usedDegree);
    }
    
    return _bspline.value().interpolateAt(time);
}
```

Reply:
Both changes together make a lot of sense :+1:.

We don't need a sentinel value, now that we have optional though `std::optional<int> findTime(double time) const;`